hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -499,14 +500,16 @@ object HoodieSparkSqlWriter {
                          latestTableSchemaOpt: Option[Schema],
                          internalSchemaOpt: Option[InternalSchema],
                          opts: Map[String, String]): Schema = {
+    val addNullForDeletedColumns = opts.getOrDefault(DataSourceWriteOptions.ADD_NULL_FOR_DELETED_COLUMNS.key(),
```

Comment:
way too many flags in this method. Let's see if we can reduce them. Irrespective, this method needs to be well-tested.

Reply:
marked RECONCILE_SCHEMA as deprecated. https://issues.apache.org/jira/browse/HUDI-6947